dhall/src/Dhall/Import.hs

```diff
@@ -580,7 +581,10 @@ loadImportWithSemanticCache
 
 
 -- Fetch encoded normal form from "semantic cache"
-fetchFromSemanticCache :: Dhall.Crypto.SHA256Digest -> IO (Maybe Data.ByteString.ByteString)
+fetchFromSemanticCache
+    :: (MonadState Bool m, MonadCatch m, MonadIO m)
```

Comment:
I think the `MonadState Bool` constraint is a bit mysterious. If we had a dedicated data type like

```
data CacheWarningState = NoWarningYet | AlreadyWarned
```

I think that would be a bit more readable.